### ISSUE TITLE
Add visualization and analysis tools for glasses

### ISSUE THREAD

**Issue by BrianGun** [microsoft/OpticSim.jl#112]
It is frequently useful to be able to plot glass properties to help in the process of choosing glasses. For example, a common plot which we don't currently support is a glass map, a scatter plot of glasses with the Abbe number along the x axis and the nominal glass index along the y axis.

It is also handy to be able to filter the glass catalog, to only show glasses from a single manufacturer, or glasses that are less expensive. A general mechanism for doing this would be valuable.

The glass catalog code is relatively short and simple so this would be a good first project for newcomers. OpticSim/src/Glasscat/utilities.jl has a few simple plotting examples which might be a helpful starting point.

**Comment by Sorn89** [comment]
Hi all

Is there a function that returns the glass id from the catalogs?

**Comment by alfredclwong** [comment]
Is this what you're looking for?

https://microsoft.github.io/OpticSim.jl/dev/glasscat/#OpticSim.GlassCat.glassid

**Comment by Sorn89** [comment]
No, I have the problem that I only get the glass names from the catalogs.But I need a way to get the data type AbstractGlass or the glass ID. So I can get nd and vd from the glass.

Maybe the problem is also due to my little Julia knowledge. But I find the project very interesting.

**Comment by BrianGun** [comment]
@friggog could you help @Sorn89?

**Comment by BrianGun** [comment]
If you want to get all the glasses in the glass catalog you can use this function:

GlassCat.AGF_GLASSES

this is a vector of glasses. If you do 

glasses = GlassCat.AGF_GLASSES

and then 

b = glasses[30] #get the 30th glass in the catalog

then

b.Nd

this will return the index of that particular glass. Not sure if this is what you were looking for. Perhaps if you are more specific we can give more directed advice.

**Comment by Sorn89** [comment]
@BrianGun Thank you very much! I was looking for such a function. This has solved my problem.
